mvpa2/cmdline/cmd_ttest.py

```diff
@@ -88,8 +122,31 @@ def run(args):
     else:
         raise ValueError('WTF you gave me? have no clue about %r' % (args.stat,))
 
+    if s.shape != out_of_mask.shape:
+        try:
+            out_of_mask = out_of_mask.reshape(s.shape)
+        except ValueError:
+            raise ValueError('Cannot use mask of shape {0} with '
+                             'data of shape {1}'.format(out_of_mask.shape, s.shape))
     s[out_of_mask] = 0
 
     verbose(1, "Saving to %s" % args.output)
-    nib.Nifti1Image(s, None, header=nis[0].header).to_filename(args.output)
+    filetype_out = guess_backend(args.output)
+    if filetype_in == 'nifti':
+        if filetype_out == 'nifti':
+            nib.Nifti1Image(s, None, header=nis[0].header).to_filename(args.output)
+        else:  # filetype_out hdf5
+            # need to get mapper and stuff
+            s_ = fmri_dataset(nis[0])
+            s_.samples = s_.a.mapper(s)
+            h5save(args.output, s_)
+    else:  # filetype_in hdf5
+        if filetype_out == 'nifti':
+            try:
+                map2nifti(dss[0], data=s).to_filename(args.output)
+            except (NameError, ValueError):
+                raise ValueError('Cannot output with requested file format')
+        else:  # filetype_out hdf5
+            s = Dataset(s, sa=dss[0].sa, fa=dss[0].fa, a=dss[0].a)
+            h5save(args.output, s)
```

Comment:
any chance to get this code tested?  ;) at least some smoke testing... so we don't blow up really badly? ;-)